pytorch_lightning/metrics/classification/utils.py

```diff
@@ -0,0 +1,385 @@
+
+    When ``num_classes`` is given, it is checked that it is consitent with input cases (binary,
+    multi-label, ...), and that, if availible, the implied number of classes in the ``C``
+    dimension is consistent with it (as well as that max label in target is smaller than it).
+
+    When ``num_classes`` is not specified in these cases, consistency of the highest target
+    value against ``C`` dimension is checked for (multi-dimensional) multi-class cases.
+
+    If ``top_k`` is larger than one, then an error is raised if the inputs are not (multi-dim)
+    multi-class with probability predictions.
+
+    Preds and target tensors are expected to be squeezed already - all dimensions should be
+    greater than 1, except perhaps the first one (N).
+
+    Args:
+        preds: tensor with predictions
+        target: tensor with ground truth labels, always integers
+        threshold:
+            Threshold probability value for transforming probability predictions to binary
+            (0,1) predictions, in the case of binary or multi-label inputs. Default: 0.5
+        num_classes: number of classes
+        is_multiclass: if True, treat binary and multi-label inputs as multi-class or multi-dim
+            multi-class with 2 classes, respectively. If False, treat multi-class and multi-dim
+            multi-class inputs with 1 or 2 classes as binary and multi-label, respectively.
+            Defaults to None, which treats inputs as they appear.
+    """
+
+    if target.is_floating_point():
+        raise ValueError("target has to be an integer tensor")
+    elif target.min() < 0:
+        raise ValueError("target has to be a non-negative tensor")
+
+    preds_float = preds.is_floating_point()
+    if not preds_float and preds.min() < 0:
+        raise ValueError("if preds are integers, they have to be non-negative")
+
+    if not preds.shape[0] == target.shape[0]:
+        raise ValueError("preds and target should have the same first dimension.")
+
+    if preds_float:
+        if preds.min() < 0 or preds.max() > 1:
+            raise ValueError(
+                "preds should be probabilities, but values were detected outside of [0,1] range"
+            )
+
+    if threshold > 1 or threshold < 0:
+        raise ValueError("Threshold should be a probability in [0,1]")
+
+    if is_multiclass is False and target.max() > 1:
+        raise ValueError("If you set is_multiclass=False, then target should not exceed 1.")
+
+    if is_multiclass is False and not preds_float and preds.max() > 1:
+        raise ValueError("If you set is_multiclass=False and preds are integers, then preds should not exceed 1.")
+
+    # Check that shape/types fall into one of the cases
+    if len(preds.shape) == len(target.shape):
+        if preds.shape != target.shape:
+            raise ValueError("if preds and target have the same number of dimensions, they should have the same shape")
```

Comment:
```suggestion
            raise ValueError(
                "preds and targets should have the same shape",
                f" got preds shape = {preds.shape} and target shape = {target.shape}."
            )
```
Just add the value, if possible, wherever an error is raised

Reply:
Cool, will add this (but using lazy string formatting)

Reply:
Eh, or without lazy string formatting - this is not logging anyway